ChangeLog

```diff
@@ -113,6 +113,8 @@
  * #2025 (The lack of Bonmin generates an error during example compilation)
  * #2031 (Multi-objective optimization example question)
  * #2032 (Documentation of NAIS)
+ * #2121 (GaussianProcess Gibbs sampling method issues)
```

Comment:
You removed the main problem (thanks!) but there are still 3 remaining tasks to close #2121 (not all of them need to be tackled in this PR if you don't want to do it).

Reply:
what are the 3 remaining?

Reply:
See the top post in #2121. I have marked the first of the 4 checkboxes as done because of this PR.

![image](https://user-images.githubusercontent.com/31989332/183845421-f7d8362d-5d9a-4811-b0b1-1bca672a88f8.png)

Reply:
* For the last point, I have no clue. Maybe @regislebrun is the good one to answer.
* For `setSamplingMethod`, ok. I will do i
* For dimension > 1, I don't remember exactly why we did it like this in the past. Now we had the paper, IMHO thereis nothing blocking for simulating fields with `output_dimension` > 1 except some data reshape

Reply:
@regislebrun @josephmure can you have a look at the fixes?

Reply:
It is great that you made the "GIBBS" method work for dimension > 1.